lib/arclight/custom_document.rb

```diff
@@ -17,5 +17,17 @@ class CustomDocument < SolrEad::Document
       t.accessrestrict(path: 'archdesc/accessrestrict/p', index_as: %i[displayable])
       t.scopecontent(path: 'archdesc/scopecontent/p', index_as: %i[displayable])
     end
+
+    def to_solr(solr_doc = {})
+      super(solr_doc)
+      Solrizer.insert_field(solr_doc, 'names', names, :facetable)
```

Comment:
We could write a long XPath expression that would do the combination. Would that be an improvement?